sdk/servicebus/service-bus/src/models.ts

```diff
@@ -4,6 +4,47 @@
 import { OperationOptionsBase } from "./modelsToBeSharedWithEventHubs";
 import Long from "long";
 
+/**
+ * Context for the error given in processError.
+ */
+export interface ProcessErrorContext {
+  /**
+   * The operation where the error originated.
+   */
+  errorSource: /**
+   * Errors that occur when autoComplete completes a message.
+   */
+  | "complete"
+    /**
+     * Errors that occur when automatic message `abandon` fails. Automatic
+     * message abandon'ing occurs when the `processMessage` handler throws an error.
+     */
+    | "abandon"
+    /**
+     * Errors thrown from the user's `processMessage` callback passed to `subscribe`
+     */
+    | "processMessageCallback"
+    /**
+     * Errors thrown when receiving messages.
+     */
+    | "receive"
+    /**
+     * Errors thrown when the user calls renewLock or when the internal lock renewer calls renewLock.
```

Comment:
`user calls renewLock` too?

Reply:
They do but that error will come back through their actual renewLock call. 

These are just for things we'd report back through processError.